src/main/java/org/jabref/gui/filewizard/FileWizardManager.java

```diff
@@ -0,0 +1,158 @@
+        // then the info file is created. Otherwise the file is deserialized.
+        if (!checkedFilesListFile.exists()) {
+            try {
+                checkedFilesListFile.createNewFile();
+                checkedFilesList = new ArrayList<>();
+            } catch (IOException ioe) {
+                LOGGER.error("couldn't create info file.");
+            }
+        } else {
+            checkedFilesList = serializer.deserializeCheckedFilesList(checkedFilesListFile);
+        }
+
+        FileWizardProgressDialog progressDialog = new FileWizardProgressDialog(stateManager.getActiveDatabase()
+                .get().getEntries().size() - checkedFilesList.size());
+
+        // The task which actually coordinates the different functionalities.
+        Task<Void> fileWizard = new Task<>() {
+            @Override
+            protected Void call() {
+                List<BibEntry> entriesList = new LinkedList<>(stateManager.getActiveDatabase().get().getEntries());
+
+                // Removes all entries which have previously been successfully linked from the list of BibEntries
+                // to be handled, in order to avoid unnecessary work for the File Wizard. The citation key is used
+                // as an identifier, because serialization doesn't work with BibEntry objects.
+                for(BibEntry entry : entriesList) {
+                    if(checkedFilesList.contains(entry.getCitationKey().get())) {
+                        Platform.runLater(() -> entriesList.remove(entry));
+                    }
+                }
+
+                // Iterates over all entries which haven't already been successfully linked.
+                for(BibEntry entry : entriesList) {
+                    if(!progressDialog.isRunning()) {
+                        for(int i = entriesList.indexOf(entry); i < entriesList.size(); i++) {
+                            unsuccessfullyLinked.add(entriesList.get(i));
+                        }
+                        return null;
+                    }
+
+                    progressDialog.updateProgress(entry, entriesList.indexOf(entry));
+
+                    LOGGER.info("File Wizard is handling " + entry.getCitationKey());
+
+                    // Checks if the linked file exists locally and if it does it copies it into the target directory.
+                    if(!fileExistsLocally(entry, directory.toPath(), stateManager.getActiveDatabase().get().getFileDirectories(preferences.getFilePreferences()))) {
+                        // Checks if the entry can be mapped to an existing file in the target directory based on the
+                        // the file's name and the conventions declared in the settings.
+                        if(!mapToFileInDirectory(entry, new AutoSetFileLinksUtil(directoryAsList, preferences.getAutoLinkPreferences(), ExternalFileTypes.getInstance()))) {
```

Comment:
How good does this scale on large tables?
As `mapToFileInDirectory()` is calling to `FindAssociatedNotLinkedFiles()`, which is a bit more compelex method, I fear this will be a bottleneck for performance.